daemon/api.go

```diff
@@ -1758,22 +1758,29 @@ func getInterfaces(c *Command, r *http.Request, user *auth.UserState) Response {
 	// Query the interface repository (this returns []*interface.Info).
 	infos := c.d.overlord.InterfaceManager().Repository().Info(opts)
 	infoJSONs := make([]*interfaceJSON, 0, len(infos))
+
 	for _, info := range infos {
 		// Convert interfaces.Info into interfaceJSON
 		plugs := make([]*plugJSON, 0, len(info.Plugs))
 		for _, plug := range info.Plugs {
+			// Re-map outgoing plug information.
```

Comment:
at this point in API if understand correctly we will possibly return "core" in some situations even if the snap doesn't exist, right?

wondering if we should go all the way and as we discussed skip this intermediate state and return "system" instead, it means that the mapping to state vs the mapping for the outgoing API world need to be different, or that we need another layer of mapping just here (core => system)

Reply:
I am looking at what it would take to return "system" to the API and "core" to to the state now.